gusto/diagnostics.py

```diff
@@ -53,7 +55,10 @@ def l2(f):
 
     @staticmethod
     def total(f):
-        return assemble(f * dx)
+        if len(f.ufl_shape) == 0:
+            return assemble(f * dx)
+        else:
+            pass
```

Comment:
I did think of having a separate list (in addition to available_diagnostics) for diagnostics that only apply to scalar fields but I wondered if it would generalise nicely to other exceptions (not that I can think of any examples!)